x-pack/plugins/reporting/server/routes/lib/authorized_user_pre_routing.test.ts

```diff
@@ -82,82 +93,101 @@ describe('authorized_user_pre_routing', function () {
   });
 
   it('should return with 401 when security is enabled and the request is unauthenticated', async function () {
-    mockCore.getPluginSetupDeps = () =>
-      ({
-        // @ts-ignore
-        ...mockCore.pluginSetupDeps,
-        security: {
-          license: { isEnabled: () => true },
-          authc: { getCurrentUser: () => null },
-        },
-      } as unknown as ReportingInternalSetup);
+    mockSetupDeps = createMockPluginSetup({
+      security: { license: { isEnabled: () => true } },
+    });
+    mockStartDeps = await createMockPluginStart(
+      { security: { authc: { getCurrentUser: () => null } } },
+      mockReportingConfig
+    );
+    mockCore = await createMockReportingCore(mockReportingConfig, mockSetupDeps, mockStartDeps);
     const mockHandler = () => {
       throw new Error('Handler callback should not be called');
     };
     const requestHandler = authorizedUserPreRouting(mockCore, mockHandler);
-    const mockResponseFactory = getMockResponseFactory();
 
-    expect(requestHandler(getMockContext(), getMockRequest(), mockResponseFactory)).toMatchObject({
+    expect(
+      await requestHandler(getMockContext(), getMockRequest(), getMockResponseFactory())
+    ).toMatchObject({
```

Comment:
nit: it's better to use `await expect(...).resolves.` because then Jest will handle exceptions/rejections properly.

```suggestion
    await expect(
      requestHandler(getMockContext(), getMockRequest(), getMockResponseFactory())
    ).resolves.toMatchObject({
```